### PR TITLE
Add layout-option to restrict rotation to portrait mode

### PR DESCRIPTION
**Pull request by Sir-Photch**
If I understand correctly, this would block the "rotate-button" to pop up every time I am lying on my side.

As suggested in: https://stackoverflow.com/questions/2730855/prevent-screen-rotation-on-android

### CODE REVIEW

**Comment by MM2-0** [comment]
It's possible that there are users that use the launcher in landscape mode (e.g. tablet users). You'd break it for them if you added `android:screenOrientation="portrait"`

Also, wrong activity, you'd need to add this to https://github.com/MM2-0/Kvaesitso/blob/main/app/ui/src/main/AndroidManifest.xml#L7-L27

**Comment by Sir-Photch** [comment]
Well, tablet screens are even more rectangular in shape. That's why I think it makes even less sense to use it in landscape mode in that case.

The rotation button is just annoying imo. I did not see that on any other launchers.

**Comment by jamal2362** [comment]
It would be better to do it through the code so that the user can decide what he wants to have.

A button in the settings where you can activate that the screen no longer rotates.

**Comment by Sir-Photch** [comment]
Well adding a switch in the settings is more complex than just modifying a manifest. I'd be willing to try this though, if anyone could point me to the right source files 😅

File: core/preferences/src/main/proto/settings.proto [hunk 1]
```diff
@@ -298,6 +298,7 @@ message Settings {
     bool bottom_search_bar = 2;
     bool reverse_search_results = 3;
     bool fixed_search_bar = 4;
+    bool fixed_rotation = 5;
```
Comment (Sir-Photch) [inline core/preferences/src/main/proto/settings.proto:301]: So I had a look on how I could add this programmatically in the layout settings. I've found that a lot of settings-methods are generated; And looking in the commit-log it seemed to me that I had to add this line for these methods to be generated. This somehow is not working though. What am I missing?

Comment (MM2-0) [reply core/preferences/src/main/proto/settings.proto:301]: They are generated at build time, either run the project or press the "build project" button in Android Studio should fix it.

Comment (Sir-Photch) [reply core/preferences/src/main/proto/settings.proto:301]: Ah, I had to remove my calls to those non-existing methods and build it for it to generate, thanks 👍🏼

File: app/ui/src/main/java/de/mm20/launcher2/ui/launcher/SharedLauncherActivity.kt [hunk 2]
```diff
@@ -132,6 +134,15 @@ abstract class SharedLauncherActivity(
                             val reverseSearchResults by viewModel.reverseSearchResults.observeAsState(false)
                             val fixedSearchBar by viewModel.fixedSearchBar.observeAsState(false)
 
+                            viewModel.fixedRotation.observe(this) { fixedRotation ->
+                                requestedOrientation = if (fixedRotation) {
+                                    ActivityInfo.SCREEN_ORIENTATION_PORTRAIT + ActivityInfo.SCREEN_ORIENTATION_NOSENSOR
```
Comment (Sir-Photch) [inline app/ui/src/main/java/de/mm20/launcher2/ui/launcher/SharedLauncherActivity.kt:139]: Assuming here that `SCREEN_ORIENTATION`-values work as flags

Comment (MM2-0) [reply app/ui/src/main/java/de/mm20/launcher2/ui/launcher/SharedLauncherActivity.kt:139]: I think so but please use bitwise or (`or`) to combine the flags

Comment (Sir-Photch) [reply app/ui/src/main/java/de/mm20/launcher2/ui/launcher/SharedLauncherActivity.kt]: Well, that did not work anyway. :)

**Comment by Sir-Photch** [comment]
I've tested it on my Pixel 7 Pro and it seems to be working as I intended. Is there anything else that needs to be added?

**Comment by MM2-0** [comment]
Looks good to me. Preference migration would be required if the default value was `true` but in that case, it's not necessary because an unset boolean defaults to `false` anyway.

**Pull request merged**